linkerd/app/core/src/errors/respond.rs

```diff
@@ -176,11 +185,20 @@ impl SyntheticHttpResponse {
         let mut rsp = http::Response::builder()
             .status(self.http_status)
             .version(version)
-            .header(http::header::CONTENT_LENGTH, "0")
-            .header(L5D_PROXY_ERROR, self.message());
+            .header(http::header::CONTENT_LENGTH, "0");
+
+        // TODO only set when client is trusted.
+        rsp = rsp.header(L5D_PROXY_ERROR, self.message());
+
+        if self.close_connection {
+            if version == http::Version::HTTP_11 {
+                rsp = rsp.header(http::header::CONNECTION, "close");
```

Comment:
why do we need to add a `connection: close` header here as well? is that intended for situations where the client isn't a linkerd proxy, or is it in case the client proxy doesn't handle the `l5d-proxy-connection` header? a comment could be nice...

Reply:
this advertises to _clients_ -- proxies and not -- that we are going to close the connection. the `l5d-proxy-connection` header advertises that the peer proxy should close the _proxied_ (server) connection.